lib/internal/Magento/Framework/Session/SessionManager.php

```diff
@@ -517,6 +549,37 @@ public function regenerateId()
         return $this;
     }
 
+    /**
+     * Regenerate session id, with saving relation between two sessions.
+     */
+    protected function regenerateSessionId()
```

Comment:
regenerateSessionId accesses the super-global variable $_SESSION.

Superglobal variables in PHP should not be accessed directly. Instead of writing:

`$name = $_POST['foo'];`
You should encapsulate the call in an single class. Most PHP frameworks already wrap Superglobals into a request object so you can use that.


https://app.codacy.com/app/shyamranpara/magento2/pullRequest?prid=1494337

Reply:
@larsroettig, thank you for the review.
The problem has been fixed.